driver-core/src/main/com/mongodb/internal/connection/DefaultServerMonitor.java

```diff
@@ -159,7 +159,7 @@ public void run() {
                     connectionPool.invalidate();
                 }
 
-                if (((connection == null || connection.supportsAdditionalTimeout())
+                if (((connection == null || shouldStreamResponses(currentServerDescription))
```

Comment:
This fixes the timeouts on Evergreen, which were caused by this bug interacting with a test bug:

* the bug here is that we have to use the same condition here as we do to decide whether streaming will be done.  Without this change, we won't stream but we also won't sleep.
* there is a bug in reactive client side encryption tests where we don't close the MongoClient after each test.  This causes us to have hundreds of open MongoClient instances 

The combination of the two means that with TLS (which won't stream, due to the TSL library issue) we will have hundreds of MongoClient instances, all with monitors polling in a tight loop.